pkgs/tools/system/acpica-tools/default.nix

```diff
@@ -27,7 +29,10 @@ stdenv.mkDerivation rec {
     "iasl"
   ];
 
-  NIX_CFLAGS_COMPILE = "-O3";
+  NIX_CFLAGS_COMPILE = [
+    "-Wno-dangling-pointer"
```

Comment:
1. why?
2. failing on Darwin.

Reply:
Apologies, extracting some version bumps from https://github.com/NixOS/nixpkgs/pull/206907 and forgot to remove gcc12 specific changes.

Removed that flag addition from this merge request.